activerecord/lib/active_record/connection_adapters/postgresql/schema_definitions.rb

```diff
@@ -191,6 +191,15 @@ def initialize(*, **)
           @unlogged = ActiveRecord::ConnectionAdapters::PostgreSQLAdapter.create_unlogged_tables
         end
 
+        def new_column_definition(name, type, **options) # :nodoc:
+          case type
+          when :virtual
```

Comment:
Has there been discussion about the history behind the `virtual` name/keyword here?

The entry in the [PostgreSQL v13 documentation](https://www.postgresql.org/docs/13/ddl-generated-columns.html) is under the "Generated Columns" heading, the SQL command itself contains `GENERATED ALWAYS AS`, and the opposite of the `STORED` flag at the end of the statement _is presumed_ to be `VIRTUAL`. As of January 2021, the `STORED` option is required.

My gut tells me that `generated` would be a more appropriate name in all the cases where this PR introduces `virtual`. What am I missing here?

Reply:
Yes, in this comment: https://github.com/rails/rails/pull/39368#issuecomment-652719159

> I think we should have the stored: true option. Postgres 12 only supports STORED for now, but this would make the t.virtual signature match the MySQL adapter and make this ready for when Postgres does eventually add the VIRTUAL option.

The idea is that using the name `virtual` with a `stored` option should:
1. Match an existing method for MySQL and make this feature more DB-agnostic
2. Support a (non-stored) "virtual" column once postgresql supports it

Reply:
Ah, thanks for clarifying @marksiemers. I was unaware of the MySQL analog.